src/pk.c

```diff
@@ -5321,7 +5321,7 @@ WOLFSSL_DSA_SIG* wolfSSL_DSA_do_sign_ex(const unsigned char* digest,
 
     WOLFSSL_ENTER("wolfSSL_DSA_do_sign_ex");
 
-    if (!digest || !dsa || inLen != WC_SHA_DIGEST_SIZE) {
+    if (!digest || !dsa || inLen != WC_SHA256_DIGEST_SIZE) {
```

Comment:
Won't this break backwards compatibly for anyone using DSA with SHA-1? Should there be a way to use either?